crates/uv-build-frontend/src/lib.rs

```diff
@@ -514,8 +517,34 @@ impl SourceBuild {
                         requirements,
                     }
                 } else {
-                    // If a `pyproject.toml` is present, but `[build-system]` is missing, proceed with
-                    // a PEP 517 build using the default backend, to match `pip` and `build`.
+                    // If a `pyproject.toml` and a `setup.py` are present, but `[build-system]` is
+                    // missing, proceed with a PEP 517 build using the default backend (setuptools),
+                    // to match `pip` and `build`. If there is no build system defined and there is
+                    // no metadata source for setuptools, fail the build instead of building a
+                    // wheel with UNKNOWN package name through setuptools.
+                    // Assumption: Setuptools can be configured through `setup.py`, `setup.cfg` or
+                    // `pyproject.toml`'s project table, otherwise it builds invalid wheels with
+                    // an UNKNOWN name.
+                    if pyproject_toml.project.is_none()
+                        && !source_tree.join("setup.py").is_file()
+                        && !source_tree.join("setup.cfg").is_file()
```

Comment:
So my concern here is... what if, in the future, setuptools supports something like `setup.toml`? Then existing versions of uv will fail here, overly-aggressively. That's partly why I was advocating that we do this "after-the-fact" (i.e., hint when we create an `UNKNOWN` wheel). How hard is that?